sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/ReversedOperatorsBase.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class ReversedOperatorsBase : SonarDiagnosticAnalyzer
```

Comment:
I'm not sure about the value of this class. Instead of:
```
SonarDiagnosticAnalyzer<-ReversedOperatorsBase <- ReversedOperatorsBase<-TUnaryExpressionSyntax><-vb/c# version
// Why not
SonarDiagnosticAnalyzer<-ReversedOperatorsBase<-TUnaryExpressionSyntax><-vb/c# version
```
(I've seen this pattern used for other rules, it's just that I don't see why there too)